25 Grammars fail to build on Windows

### Summary

Helix can fetch grammars, and will build with the `HELIX_DISABLE_AUTO_GRAMMAR_BUILD` set to `true`, however 25 grammars fail to build both during the installation of helix and seperately via `hx --grammar build`. Here is the output from the console, starting from the end of the successful compilations:

```powershell
  124 grammars built now
        ["awk", "bass", "bibtex", "bicep", "c", "c-sharp", "cairo", "capnp", "clojure", "comment", "cpon", "css", "cue", "d", "dart", "devicetree", "dhall", "diff", "dockerfile", "dot", "dtd", "edoc", "eex", "elvish", "embedded-template", "erlang", "esdl", "fish", "git-commit", "git-config", "git-rebase", "gitattributes", "gitignore", "gleam", "glsl", "go", "godot-resource", "gomod", "gotmpl", "gowork", "graphql", "hare", "haskell", "heex", "hosts", "hurl", "iex", "ini", "java", "javascript", "jsdoc", "json", "jsonnet", "julia", "kdl", "kotlin", "latex", "ledger", "llvm", "llvm-mir", "lua", "make", "markdoc", "markdown", "markdown_inline", "matlab", "mermaid", "meson", "nasm", "nix", "nu", "odin", "opencl", "openscad", "pascal", "passwd", "pem", "po", "ponylang", "prisma", "protobuf", "prql", "qmljs", "r", "regex", "rego", "rescript", "robot", "rst", "rust", "scala", "scheme", "scss", "slint", "smithy", "sml", "solidity", "sql", "sshclientconfig", "svelte", "sway", "swift", "tablegen", "task", "toml", "tsq", "tsx", "twig", "typescript", "ungrammar", "uxntal", "v", "vala", "verilog", "vhdl", "vhs", "wast", "wat", "wgsl", "wit", "xit", "xml", "yuck", "zig"]
  Failure 1/25: html Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\html_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\html_scanner.o', object file assumed

  Failure 2/25: nickel Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\nickel_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\nickel_scanner.o', object file assumed

  Failure 3/25: python Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\python_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\python_scanner.o', object file assumed

  Failure 4/25: beancount Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\beancount_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\beancount_scanner.o', object file assumed

  Failure 5/25: cmake Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\cmake_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\cmake_scanner.o', object file assumed

  Failure 6/25: vue Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\vue_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\vue_scanner.o', object file assumed

  Failure 7/25: yaml Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\yaml_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\yaml_scanner.o', object file assumed

  Failure 8/25: php Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\php_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\php_scanner.o', object file assumed

  Failure 9/25: bash Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\bash_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\bash_scanner.o', object file assumed

  Failure 10/25: elixir Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\elixir_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\elixir_scanner.o', object file assumed

  Failure 11/25: elm Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\elm_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\elm_scanner.o', object file assumed

  Failure 12/25: hcl Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\hcl_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\hcl_scanner.o', object file assumed

  Failure 13/25: cpp Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\cpp_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\cpp_scanner.o', object file assumed

  Failure 14/25: org Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\org_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\org_scanner.o', object file assumed

  Failure 15/25: gdscript Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\gdscript_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\gdscript_scanner.o', object file assumed

  Failure 16/25: ocaml-interface Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\ocaml-interface_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\ocaml-interface_scanner.o', object file assumed

  Failure 17/25: ruby Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\ruby_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\ruby_scanner.o', object file assumed

  Failure 18/25: ocaml Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\ocaml_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\ocaml_scanner.o', object file assumed
  Failure 19/25: astro Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\astro_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\astro_scanner.o', object file assumed

  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\fortran_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\fortran_scanner.o', object file assumed

  Failure 21/25: cabal Parser compilation failed.
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\cabal_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\cabal_scanner.o', object file assumed

  Failure 22/25: just Parser compilation failed.
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\just_scanner.o'

  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\just_scanner.o', object file assumed

  Failure 23/25: perl Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\perl_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\perl_scanner.o', object file assumed

  Failure 24/25: nim Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\nim_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\nim_scanner.o', object file assumed

  Failure 25/25: lean Parser compilation failed.
  Stdout: parser.c
  LINK : fatal error LNK1181: cannot open input file 'D:\GitHub\helix\runtime\grammars\lean_scanner.o'

  Stderr: cl : Command line warning D9025 : overriding '/O2' with '/Od'
  cl : Command line warning D9024 : unrecognized source file type 'D:\GitHub\helix\runtime\grammars\lean_scanner.o', object file assumed


  --- stderr
  thread 'main' panicked at 'Failed to compile tree-sitter grammars: 25 grammars failed to build', helix-term\build.rs:7:14
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
error: failed to compile `helix-term v0.6.0 (D:\GitHub\helix\helix-term)`, intermediate artifacts can be found at `D:\GitHub\helix\target`
```

### Reproduction Steps

I tried this:

1. Navigated to the local helix repo directory and ran `cargo install --path helix-term --locked`
2. Installed helix as above but with `HELIX_DISABLE_AUTO_GRAMMAR_BUILD` set to `true`, then ran `hx --grammar fetch` and `hx --grammar build`

### Helix log

_No response_

### Platform

Windows 10

### Terminal Emulator

Powershell in Windows Terminal

### Helix Version

helix 23.03 (a1044a6)

--- a/helix-loader/src/grammar.rs
+++ b/helix-loader/src/grammar.rs
@@ -433,11 +433,12 @@ fn build_tree_sitter_library(
     for (key, value) in compiler.env() {
         command.env(key, value);
     }
+
     command.args(compiler.args());
     // used to delay dropping the temporary object file until after the compilation is complete
     let _path_guard;
 
-    if cfg!(all(windows, target_env = "msvc")) {
+    if compiler.is_like_msvc() {
         command
             .args(["/nologo", "/LD", "/I"])
             .arg(header_path)
@@ -455,20 +456,20 @@ fn build_tree_sitter_library(
                 }
                 cpp_command.args(compiler.args());
                 let object_file =
-                    library_path.with_file_name(format!("{}_scanner.o", &grammar.grammar_id));
+                    library_path.with_file_name(format!("{}_scanner.obj", &grammar.grammar_id));
                 cpp_command
                     .args(["/nologo", "/LD", "/I"])
                     .arg(header_path)
                     .arg("/Od")
                     .arg("/utf-8")
                     .arg("/std:c++14")
-                    .arg("/o")
-                    .arg(&object_file)
+                    .arg(format!("/Fo{}", object_file.display()))
                     .arg("/c")
                     .arg(scanner_path);
                 let output = cpp_command
                     .output()
                     .context("Failed to execute C++ compiler")?;
+
                 if !output.status.success() {
                     return Err(anyhow!(
                         "Parser compilation failed.\nStdout: {}\nStderr: {}",

